Remaining proxy auth help articles (#65)
logit-io/docs-content: src/pages/infrastructure-metrics/visualizers/api.mdx

---
title: Access Grafana features via REST API
metaTitle: Learn How to Access Grafana features via REST API
description: Read Logit.io’s help article to learn how to access Grafana features via REST API.
---

# Accessing Grafana features via REST API

You can use Basic Authentication to connect to the Logit.io Proxy and access Grafana without having to go 
through the Logit.io Dashboard. This can improve your user experience and allow proper integration between your company, your applications 
and the Logit.io service.

> You can, if you choose to, connect your application directly to Logit.io to do a range of things such as saving dashboards, creating users, 
> and updating data sources etc. or access Grafana straight from your computer. 

Reviewer: ```suggestion
You have the option to connect your application directly to the Logit.io API to do a range of things such as saving dashboards, creating users, updating data sources or access Grafana straight from your computer.
```

This article explains how to call REST API Endpoints to access Grafana using our Logit.io proxy and connect to it using Basic 
Authentication.

<Callout type="info">
    Basic Authentication is a simple authentication method in HTTP(S) where the client includes a username and password with each request. 
    The username and password is combined with a colon in-between as a string. You don't need to worry about how to create a basic authentication 
    header as we have done it for you. 
</Callout>

Each request to a REST API Endpoint must include all of the necessary information for Grafana to fulfill the request. If all the necessary 
information is provided the API requests return JSON output.

Use the following operations to interact with Grafana via the Logit.io Proxy.

> * __GET__ - Fetches the information.
> * __POST__ - Adds new information.
> * __PUT__ - Updates the existing information.
> * __PATCH__ - Applies partial modifications to the existing information.
> * __DELETE__ - Removes the information.

Reviewer: ```suggestion
* __GET__ - Fetches the data.
* __POST__ - Adds new data.
* __PUT__ - Updates the existing data.
* __PATCH__ - Applies partial modifications to the existing data.
* __DELETE__ - Removes the data.
```
